File: iosApp/iosApp/ui/components/SecondaryButton.swift
import SwiftUI

/// Outlined secondary button — white surface, 1.5px border, brand text.
/// Wraps SwiftUI's native `Button`.
struct SecondaryButton: View {
    let label: String
    let action: () -> Void
    var enabled: Bool = true

    @Environment(\.aromexColors) private var colors
    @Environment(\.aromexTypography) private var typography
    @Environment(\.aromexDimensions) private var dimensions

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(typography.button)
                .frame(maxWidth: .infinity)
                .frame(height: dimensions.buttonHeight)
                .foregroundStyle(colors.textPrimary)
                .background(
                    RoundedRectangle(cornerRadius: dimensions.radiusButton)
                        .fill(colors.surface)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: dimensions.radiusButton)
                        .stroke(colors.border, lineWidth: dimensions.borderField)
                )
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

File: androidApp/src/main/kotlin/com/humblesolutions/aromex/ui/components/SecondaryButton.kt
package com.humblesolutions.aromex.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.humblesolutions.aromex.ui.theme.AromexTheme

/**
 * Outlined secondary button — white surface, 1.5px brand-border, brand text.
 * Wraps Material 3's [OutlinedButton] so it inherits ripple + accessibility.
 */
@Composable
fun SecondaryButton(
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    val colors = AromexTheme.colors
    val dimensions = AromexTheme.dimensions

    OutlinedButton(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .height(dimensions.buttonHeight),
        enabled = enabled,
        shape = RoundedCornerShape(dimensions.radiusButton),
        border = BorderStroke(dimensions.borderField, colors.border),
        colors = ButtonDefaults.outlinedButtonColors(
            containerColor = colors.surface,
            contentColor = colors.textPrimary,
            disabledContentColor = colors.textTertiary,
        ),
    ) {
        Text(text = label, style = AromexTheme.typography.button)
    }
}
